drv/sidecar-seq-server/src/main.rs

```diff
@@ -477,11 +507,30 @@ impl idl::InOrderSequencerImpl for ServerImpl {
             .as_ref()
             .ok_or(SeqError::NoFrontIOBoard)?;
 
-        front_io_board
+        match front_io_board
             .phy()
-            .set_osc_good(good)
+            .osc_good()
             .map_err(SeqError::from)
-            .map_err(RequestError::from)
+            .map_err(RequestError::from)?
+        {
+            // The oscillator is already marked good, ignore this call.
+            Some(true) => Ok(()),
```

Comment:
This seems to make it impossible to transition from `good = true` → `good = false`, since we only call `set_osc_good(good)` in the other case.  Is that desired?

Reply:
That is intended, yes. Because the osc good/bad state can only change when the oscillator is power cycled. This in turn can only happen if the whole front IO board is power cycled which then causes the front IO FPGAs to be reset erasing the value of that register. So once `monorail` has determined once that the oscillator is good (because the link is up) this value shouldn't be changed..

Reply:
Okay – in that case, maybe we should remove the `good: bool` from this function entirely, since it's a bit misleading.

Then, we'd call the function conditionally [here](https://github.com/oxidecomputer/hubris/blob/reduce_monorail_restart/task/monorail-server/src/bsp/sidecar_bc.rs#L231), something like
```rust
            // Notify the sequencer when the oscillator becomes good.  Once
            // the oscillator is good any future resets of the PHY do not
            // require a full power cycle of the front IO board.
            if osc_good {
                self.seq
                    .set_front_io_phy_osc_good()
                    .map_err(|e| VscError::ProxyError(e.into()))?;
            } else {
                ringbuf_entry!(Trace::FrontIoPhyOscillatorBad)
            }
```

Reply:
I added this explanation to the comment.

Reply:
We spoke offline about the implications of having a three-state variable in the front IO board restart logic. This has been refactored a bit into an explicit type so the logic dealing with the `Unknown` state is maybe a bit more readable.